core/pta/apdu.c

```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/*
+ * Copyright (C) Foundries Ltd. 2021 - All Rights Reserved
+ * Author: Jorge Ramirez <jorge@foundries.io>
+ */
+#include <crypto/crypto_secure_element.h>
+#include <kernel/pseudo_ta.h>
+#include <pta_apdu.h>
+
+#define PTA_NAME "pta.apdu"
+
+static TEE_Result invoke_command(void *session_context __unused,
+				 uint32_t command_id, uint32_t pt,
+				 TEE_Param params[TEE_NUM_PARAMS])
+{
+	const uint32_t exp_pt = TEE_PARAM_TYPES(TEE_PARAM_TYPE_VALUE_INPUT,
+						TEE_PARAM_TYPE_MEMREF_INPUT,
+						TEE_PARAM_TYPE_MEMREF_INPUT,
+						TEE_PARAM_TYPE_MEMREF_OUTPUT);
+	TEE_Result ret = TEE_ERROR_NOT_IMPLEMENTED;
+	size_t len = 0;
+
+	FMSG("command entry point for pseudo-TA \"%s\"", PTA_NAME);
+
+	if (pt != exp_pt)
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	switch (command_id) {
+	case PTA_CMD_TXRX_APDU_RAW_FRAME:
+		len = params[3].memref.size;
+		ret = crypto_secure_element_do_apdu(params[0].value.a,
```

Comment:
This reuses straight values of `PTA_APDU_TXRX_CASE_*` as values of `CRYPTO_APDU_CASE_*`. Consistency should be always asserted at build time.

The point is on one hand PTA API must define explicit values (0, 1, ...) for these IDs, on the other hand 
`enum crypto_apdu_type` from _crypto_secure_element.h_ cannt rely on the PTA IDs.
So either a conversion, or value compliance + assertion at build time.

Reply:
seems an overkill at this level of abstraction but ok..